nilearn/glm/contrasts.py

```diff
@@ -525,15 +527,30 @@ def compute_fixed_effects(
             f"from the number of variance images ({len(variance_imgs)})."
         )
 
-    if isinstance(mask, NiftiMasker):
+    if isinstance(mask, (NiftiMasker, SurfaceMasker)):
         masker = mask.fit()
     elif mask is None:
-        masker = NiftiMasker().fit(contrast_imgs)
+        if isinstance(contrast_imgs[0], SurfaceImage):
+            surf_data = {}
+            for part in contrast_imgs[0].mesh.parts:
+                surf_data[part] = np.ones(
+                    contrast_imgs[0].data.parts[part].shape[0], dtype=bool
+                )
+            mask_img = SurfaceImage(mesh=contrast_imgs[0].mesh, data=surf_data)
+            masker = SurfaceMasker(mask_img=mask_img).fit(contrast_imgs[0])
+        else:
+            masker = NiftiMasker().fit(contrast_imgs)
+    elif isinstance(mask, SurfaceImage):
+        masker = SurfaceMasker().fit(contrast_imgs[0])
```

Comment:
```suggestion
        if isinstance(contrast_imgs[0], SurfaceImage):
            masker = SurfaceMasker().fit(contrast_imgs[0])
        else:
            masker = NiftiMasker().fit(contrast_imgs)
    elif isinstance(mask, SurfaceImage):
        masker = SurfaceMasker(mask_img=mask).fit(contrast_imgs[0])
```

Given how the surface masker works (estimates mask from images passed at fit time it no mask was provided at instantiation) I think we can simplify this way.

Does it make sense?